routing/payment_lifecycle.go

```diff
@@ -456,32 +493,18 @@ func (p *shardHandler) collectResultAsync(attempt *channeldb.HTLCAttemptInfo) {
 					attempt.AttemptID, p.identifier, err)
 			}
 
-			select {
-			case p.shardErrors <- err:
-			case <-p.router.quit:
-			case <-p.quit:
-			}
+			// Overwrite errToSend and return.
```

Comment:
May be worth mentioning here (to make the control flow clearer) that we're just setting the "return param" for the `defer` above (`handleResultErr`).